### ISSUE TITLE
Solar "To grid" doesn't account for subpanel consumption

### ISSUE THREAD

**Issue by n7qnm** [magico13/ha-emporia-vue#254]
I have a somewhat unusual situation - we have a solar array  with ONE meter feeding TWO panels.  The solar array feeds into ONE of the two panels.   I have  emporia taps on the inputs to both panels,

The solar input can exceed the draw from panel 1, and feed power back OUT of panel 1 toward the meter; but, before it gets to the meter and actually goes to the grid, it's consumed by panel 2 until/unless the solar power exceeds the draw from BOTH panels.

So, the graphs from this wind up OVERSTATING my "to grid" value.

Is there a way this could modified to take into account multiple consumers and actually do a "net" calculation (as opposed to just working from the taps on the one "mains"

TIA!

Clay Jackson

**Comment by n7qnm** [comment]
OK – but in my setup, I have 2 electrical panels, something like this:

 

 

                      |-------Clamps 2-----Panel 2-------(Other Loads)

Meter-------|

                      |-------Clamps 1-----Panel 1-------(Loads)

                                                        |

                                                        Clamps

                                                           |

                                                  Solar array

 

I have TWO Emporia boxes – one in Panel 1 and one in Panel 2, because I want to see ALL of my consumption.   For example, at least according to my electrician,

 

If the Solar array is producing 3 KW and panel 1 is drawing 1KW, clamps 1 will read 2KW “to grid”.  However, if panel 2 is ALSO drawing 1 KW, what’s REALLY going to the grid is only 1KW.

 

It appears that Emporia accounts for this on their website.

**Comment by Esprit1st** [comment]
Yeah, again, your setup is incorrect. You need to have only one set up "grid clamps" and that should be on panel 1. Because that's the only clamps that measure power going to the grid. You are currently measuring panel 2 to grid as to grid which it isn't. It's only to panel 1. Which if it isn't consuming everything happens also to go to grid. However you don't really have to change anything. Just ignore the panel 2 to grid in HA. it doesn't give you any useful information anyways. Panel 1 to grid is your energy exported and all other clamps are your users and solar generation. You basically treat both emporia as one.

**Comment by richardcalgary** [comment]
@n7qnm is your panel 2 actually fed from panel 1, or do you have a splitter in your meter panel?  If panel 2 is fed from panel 1 you should configure it as a nested panel in the emporia configuration.

**Comment by richardcalgary** [comment]
In your case you will have one panel showing export to grid, and the other panel showing import from the grid. If you add up the two panel loads you should get the net to/from the grid.  I have 2 panels each with solar panels on them and a vue gen 2. Only one panel is tied to the meter, with the other panel being nested in the emporia configuration.  As I understand it the vue gen 2 does not integrate separate panels, you would have to do the calculation in a template sensor.

From app:
***@***.***
